src/cli/common.rs

```diff
@@ -381,70 +379,26 @@ where
 }
 
 pub(super) fn list_items(
-    distributable: DistributableToolchain<'_>,
-    f: impl Fn(&ComponentStatus) -> Option<&str>,
+    items: impl Iterator<Item = (String, bool)>,
     installed_only: bool,
     quiet: bool,
     process: &Process,
 ) -> Result<utils::ExitCode> {
     let mut t = process.stdout().terminal(process);
-    for component in distributable.components()? {
-        let Some(name) = f(&component) else { continue };
-        match (component.available, component.installed, installed_only) {
-            (false, _, _) | (_, false, true) => continue,
-            (true, true, false) if !quiet => {
+    for (name, installed) in items {
+        if !installed_only || installed {
+            if installed && !installed_only && !quiet {
```

Comment:
Nit: How about using `if installed && !installed_only && !quiet { .. } else if installed || !installed_only { .. }` to prevent deep nesting?